client/jetpack-connect/happychat-button.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import HappychatButton from 'components/happychat/button';
+import HappychatConnection from 'components/happychat/connection';
+import {
+	hasActiveHappychatSession,
+	isHappychatAvailable,
+} from 'state/happychat/selectors';
+
+const JetpackConnectHappychatButton = ( {
+	children,
+	isChatActive,
+	isChatAvailable,
+	translate,
+} ) => {
+	if ( ! isChatAvailable && ! isChatActive ) {
+		return (
+			<div>
+				<HappychatConnection />
+				{ children }
```

Comment:
It's actually not a common practice, but it'll be a handy way to handle our Happychat use cases within Jetpack Connect. The idea is: either render the button, or render the children (which basically serve as a fallback in case Happychat is not available).